File: seed/php-sdk/allof-inline/src/Types/UserSearchResponse.php
<?php

namespace Seed\Types;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;
use Seed\Core\Types\ArrayType;

class UserSearchResponse extends JsonSerializableType
{
    /**
     * @var PagingCursors $paging
     */
    #[JsonProperty('paging')]
    public PagingCursors $paging;

    /**
     * @var ?array<User> $results Current page of results from the requested resource.
     */
    #[JsonProperty('results'), ArrayType([User::class])]
    public ?array $results;

    /**
     * @param array{
     *   paging: PagingCursors,
     *   results?: ?array<User>,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->paging = $values['paging'];
        $this->results = $values['results'] ?? null;
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}

File: seed/rust-sdk/allof/src/api/types/user_search_response.rs
pub use crate::prelude::*;

#[derive(Debug, Clone, Serialize, Deserialize, Default, PartialEq, Eq, Hash)]
pub struct UserSearchResponse {
    /// Current page of results from the requested resource.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub results: Option<Vec<User>>,
    #[serde(default)]
    pub paging: PagingCursors,
}

impl UserSearchResponse {
    pub fn builder() -> UserSearchResponseBuilder {
        <UserSearchResponseBuilder as Default>::default()
    }
}

#[derive(Clone, PartialEq, Default, Debug)]
#[non_exhaustive]
pub struct UserSearchResponseBuilder {
    results: Option<Vec<User>>,
    paging: Option<PagingCursors>,
}

impl UserSearchResponseBuilder {
    pub fn results(mut self, value: Vec<User>) -> Self {
        self.results = Some(value);
        self
    }

    pub fn paging(mut self, value: PagingCursors) -> Self {
        self.paging = Some(value);
        self
    }

    /// Consumes the builder and constructs a [`UserSearchResponse`].
    /// This method will fail if any of the following fields are not set:
    /// - [`paging`](UserSearchResponseBuilder::paging)
    pub fn build(self) -> Result<UserSearchResponse, BuildError> {
        Ok(UserSearchResponse {
            results: self.results,
            paging: self
                .paging
                .ok_or_else(|| BuildError::missing_field("paging"))?,
        })
    }
}
